bcipy/helpers/system_utils.py

```diff
@@ -70,23 +72,45 @@ def bcipy_version() -> str:
     return f'{version} - {sha_hash}' if sha_hash else version
 
 
-def get_system_info() -> dict:
-    """Get System Information.
-    """
+def get_screen_resolution() -> Tuple[int, int]:
+    """Gets the screen resolution.
+
+    Note: Use this method if only the screen resolution is needed; it is much more efficient
+    than extracting that information from the dict returned by the get_system_info method.
+
+    Returns
+    -------
+        (width, height)
+     """
+    screen = pyglet.canvas.get_display().get_default_screen()
+    return (screen.width, screen.height)
 
-    # Three lines for getting screen resolution
-    platform = pyglet.window.get_platform()
-    display = platform.get_default_display()
-    screen = display.get_default_screen()
 
-    mem = psutil.virtual_memory()
+def get_system_info() -> dict:
+    """Get System Information.
+    See: https://stackoverflow.com/questions/3103178/how-to-get-the-system-info-with-python
 
+    Returns
+    -------
+        dict of system-related properties, including ['os', 'py_version', 'resolution', 
+          'memory', 'bcipy_version', 'platform', 'platform-release', 'platform-version', 
+          'architecture', 'processor', 'cpu_count', 'hz', 'ram']
+    """
+    screen_width, screen_height = get_screen_resolution()
     return {
         'os': sys.platform,
         'py_version': sys.version,
-        'resolution': [screen.width, screen.height],
-        'memory': mem.available / 1024 / 1024,
-        'bcipy_version': bcipy_version()
+        'resolution': [screen_width, screen_height],
+        'memory': psutil.virtual_memory().available / 1024 / 1024,
+        'bcipy_version': bcipy_version(),
+        'platform': platform.system(),
+        'platform-release': platform.release(),
+        'platform-version': platform.version(),
+        'architecture': platform.machine(),
+        'processor': platform.processor(),
+        'cpu_count': os.cpu_count(),
```

Comment:
It looks like we can get the CPU name in most cases. It would be worth adding! https://github.com/workhorsy/py-cpuinfo#raw-fields